mso/resource_mso_schema_template_contract.go

```diff
@@ -48,511 +50,515 @@ func resourceMSOTemplateContract() *schema.Resource {
 				Optional: true,
 				Computed: true,
 			},
-			"filter_type": &schema.Schema{
+			"scope": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
 				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"application-profile",
+					"tenant",
+					"context",
+					"global",
+				}, false),
 			},
-			"scope": &schema.Schema{
+			"target_dscp": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
 				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"af11",
+					"af12",
+					"af13",
+					"af21",
+					"af22",
+					"af23",
+					"af31",
+					"af32",
+					"af33",
+					"af41",
+					"af42",
+					"af43",
+					"cs0",
+					"cs1",
+					"cs2",
+					"cs3",
+					"cs4",
+					"cs5",
+					"cs6",
+					"cs7",
+					"expeditedForwarding",
+					"voiceAdmit",
+					"unspecified",
+				}, false),
 			},
-			"filter_relationships": {
-				Type: schema.TypeMap,
+			"priority": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"level6",
+					"level5",
+					"level4",
+					"level3",
+					"level2",
+					"level1",
+					"unspecified",
+				}, false),
+			},
+			"filter_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"bothWay",
+					"oneWay",
+				}, false),
+			},
+			"filter_relationship": {
```

Comment:
Why did we flip filter_relationship and filter_relationships order in schema? This breaks the diff calculation quite a lot.